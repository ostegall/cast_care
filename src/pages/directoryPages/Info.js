import { Text, View, Pressable } from 'react-native';
import styles from '../../styles';

export default function Info({navigation}) { //Page for determining which type of hospital user would like to find nearest to them
    return (
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>  
        <Text styles={styles.page_text}>
          Select which type of location you want to find
        </Text>
        <Pressable style={styles.button}
              onPress= {() => {
                navigation.navigate("UCInfo") //Navigates to Urgent Care Page
              }}
        >
          <Text style={styles.text}>
            Urgent Care
          </Text>
        </Pressable>
        <Pressable style={styles.button}
              onPress= {() => {
                navigation.navigate("EDInfo") //Navigates to Emergency Department Page
              }}
        >
          <Text style={styles.text}>
            Emergency Department
          </Text>
        </Pressable>
      </View>
    )
  }