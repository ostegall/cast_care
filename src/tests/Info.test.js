import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import Info from '../pages/directoryPages/Info';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<Info />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<Info />).toJSON();
  expect(tree).toMatchSnapshot();
});