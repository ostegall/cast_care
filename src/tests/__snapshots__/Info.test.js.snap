// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 23,
          "fontWeight": "bold",
          "letterSpacing": 0.5,
          "textAlign": "center",
        }
      }
    >
      Find your nearest UNC Hospital
    </Text>
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      Where would you like to go?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "height": 83,
          "marginLeft": 24,
          "marginRight": 23,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Urgent Care
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Emergency Department
        </Text>
      </View>
    </View>
  </View>
</View>
`;
